.github/workflows/reproducible_build.yml

```diff
@@ -0,0 +1,33 @@
+name: Reproducible-build
+
+on:
+  push:
+    branches: [ main, 'release-*' ]
+  pull_request:
+    branches: [ main, 'release-*' ]
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  reproducible-build:
+    runs-on: [self-hosted, prod, Linux, X64, docker]
+    steps:
+      # This is a workaround from: https://github.com/actions/checkout/issues/590#issuecomment-970586842
+      - run: "git checkout -f $(git -c user.name=x -c user.email=x@x commit-tree $(git hash-object -t tree /dev/null) < /dev/null) || :"
+      - uses: actions/checkout@v3
+        with:
+          # Recursive checkout required to build Solidity contracts and bindings.
+          submodules: recursive
+      - uses: ./.github/actions/rustup
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: echo "expected_image_id=9ee1612b0a7e270f8df248e47dc85d9908ff4c1e7df42f398a65ca878b57a23d" >> $GITHUB_ENV
+      - run: echo "branch_image_id=$(echo -n $(cargo run --bin cargo-risczero -- risczero build --manifest-path examples/factors/methods/guest/Cargo.toml 2>&1) | rev | cut -d ' ' -f 1 | rev )" >> $GITHUB_ENV
```

Comment:
It'd be great if we could make this a rust unit or integration test so that devs locally can just do `cargo test` to see if we broke the docker stuff.

Reply:
Added 👍